blockchain/state_manager/src/utils.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::errors::*;
+use crate::StateManager;
+use actor::miner::State;
+use address::{Address, Payload};
+use blockstore::BlockStore;
+use cid::Cid;
+use fil_types::{RegisteredProof, SectorInfo, SectorSize};
+use filecoin_proofs_api::{post::generate_winning_post_sector_challenge, ProverId};
+use std::convert::TryInto;
+
+pub fn get_sectors_winning_for_winning_post<DB>(
+    state_manager: &StateManager<DB>,
+    st: &Cid,
+    address: &Address,
+    rand: &[u8; 32],
+) -> Result<Vec<SectorInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: State = state_manager.load_actor_state(&address, &st)?;
+    let mut not_proving = miner_actor_state
+        .faults
+        .clone()
+        .merge(&miner_actor_state.recoveries)
+        .map_err(|_| Error::Other("Could not merge bitfield".to_string()))?;
+    let sector_set =
+        miner_actor_state.load_sector_infos(&*state_manager.get_block_store(), &mut not_proving)?;
+    if sector_set.is_empty() {
+        return Ok(Vec::new());
+    }
+    let seal_proof_type = match miner_actor_state.info.sector_size {
+        SectorSize::_2KiB => RegisteredProof::StackedDRG2KiBSeal,
+        SectorSize::_8MiB => RegisteredProof::StackedDRG8MiBSeal,
+        SectorSize::_512MiB => RegisteredProof::StackedDRG512MiBSeal,
+        SectorSize::_32GiB => RegisteredProof::StackedDRG32GiBSeal,
+        SectorSize::_64GiB => RegisteredProof::StackedDRG64GiBSeal,
+    };
+    let wpt = seal_proof_type.registered_winning_post_proof()?;
+
+    let mid = match address.payload() {
+        Payload::ID(new_id) => Address::new_id(*new_id),
+        _ => return Err(Error::Other(format!("getting miner id {:}", address))),
+    };
```

Comment:
same with this as before, you don't need to reconstruct the address, they only get the uint from the id protocol for ffi interactions